RELEASING.md

```diff
@@ -0,0 +1,148 @@
+# OpenTelemetry Release Process
+
+## Tagging the Release
+
+The first step in the release process is to create a release branch, bump versions, and create a tag
+for the release. Our release branches follow the naming convention of v<major>.<minor>.x, while the
+tags include the patch version v<major>.<minor>.<patch>. For example, the same branch v0.3.x would
+be used to create all v0.3.* tags (e.g. v0.3.0, v0.3.1).
```

Comment:
This sounds great to me, and it sounds we might use the same pattern on other languages, btw (as a recommendation, and if it applies, of course).

Reply:
yes, already applied this to proto :). Please push this to python :))